api/layer/tree_service.go

```diff
@@ -74,11 +74,10 @@ type TreeService interface {
 	// If object id to remove is not found returns ErrNoNodeToRemove error.
 	AddPart(ctx context.Context, bktInfo *data.BucketInfo, multipartNodeID uint64, info *data.PartInfo) (oldObjIDToDelete oid.ID, err error)
 	GetParts(ctx context.Context, bktInfo *data.BucketInfo, multipartNodeID uint64) ([]*data.PartInfo, error)
-	// GetLastPart returns the latest uploaded part.
-	//
+	// GetPartByNumber returns the part by number. If part was uploaded few times the newest one will be returned.
 	// Return errors:
 	//   - [ErrPartListIsEmpty] if there is no parts in the upload id.
-	GetLastPart(ctx context.Context, bktInfo *data.BucketInfo, multipartNodeID uint64) (*data.PartInfo, error)
+	GetPartByNumber(ctx context.Context, bktInfo *data.BucketInfo, multipartNodeID uint64, number int) (*data.PartInfo, error)
```

Comment:
Why this change? I don't see you requesting anything but the last part.

Reply:
but how is it related to V1 -> V2 move?

Reply:
it was expected to simplify the process. can you, please, describe a little why it is required to _now_ and was not needed _then_?

Reply:
> V1 has only previous link

yes, but if you change the first one, you have to change `previous` for every next file, cause every next part should be recalculated... but it wasn't like that before?

Reply:
also, i saw that the last part is always zero-length (in NeoFS terms), can we put the first one with an empty payload too just to fix the first child ID? it will not be reuploaded then ever. @roman-khimov

Reply:
https://github.com/nspcc-dev/neofs-s3-gw/pull/957#discussion_r1634639794

Reply:
whoops, missed that, i am totally "for" uploading the first object as an initial multipart uploading ID and fixing this important split part

Reply:
So, do we still need this `number` parameter?

Reply:
Sure, we still have to take the previous part to take hashes from them. Just take the latest part isn't helpful for us

Reply:
But `GetLastPart` did exactly that without `number`, no?

Reply:
In common case of course, in reuploading case no
For instance: we have parts [1,2,3]. We are reuploading part 2. `GetLastPart` will return 3, which is not valid. GetPartByNumber will return 1 (because we take currentPartID - 1) - as the last stable point